src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    // The VS LSP client supports streaming using IProgress<T> on various requests.
+    // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
```

Comment:
@dibarbet @allisonchou aren't we in 16.6?  does this ocmment still apply?

Reply:
Yep! this should be fixed in master-vs-deps (it requires 16.6 package versions).  This comment should no longer apply.

Reply:
Can you walk @allisonchou through how to use IProgress then?  (or point her at an example that does that)?

Reply:
Sure - I believe the plan was to first move this to the actual LSP client, then do a followup with the streaming changes.  felt that it might be easier to split them up to reduce the complexity (in writing and reviewing)

@allisonchou ping me if you have any questions.  The current example we have is here - https://github.com/dotnet/roslyn/blob/master-vs-deps/src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs#L125 which we'd probably need to hook up to the findusages context

Reply:
i'm find with that.  let's get this in asap, then we can do streaming :)